KubeArmor/core/kubeUpdate.go

```diff
@@ -606,14 +609,15 @@ func (dm *KubeArmorDaemon) WatchK8sPods() {
 							appArmorAnnotations[container.Name] = "kubearmor-" + pod.Metadata["namespaceName"] + "-" + podOwnerName + "-" + container.Name
 							updateAppArmor = true
 						}
+						containersCount += 1
 					}
 
 					if event.Type == "ADDED" {
 						// update apparmor profiles
 						dm.RuntimeEnforcer.UpdateAppArmorProfiles(pod.Metadata["podName"], "ADDED", appArmorAnnotations)
 
 						if updateAppArmor && pod.Annotations["kubearmor-policy"] == "enabled" {
-							if deploymentName, ok := pod.Metadata["deploymentName"]; ok {
+							if deploymentName, ok := pod.Metadata["deploymentName"]; ok && containersCount != appArmorAnnotationsCount {
```

Comment:
We only patch deployment if annotation don't exist already?

Reply:
yes. If the annotation does not exist or it is uncomplete then we need to patch the deployment as reaching this condition means that the annotation controller is not healthy.